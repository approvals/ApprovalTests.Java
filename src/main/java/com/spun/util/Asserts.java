package com.spun.util;

import java.io.File;

public class Asserts
{
  public static void t()
  {
    // can you see it now?  good.
    // how about this?  
    // kind of slow...
  }
  /***********************************************************************/
  public static <T> T assertNotNull(String label, T o)
  {
    if (o == null) { throw new NullPointerException(label + " was null"); }
    return o;
  }
  /***********************************************************************/
  public static void assertEqual(String label, int expected, int actual)
  {
    if (expected != actual) { throw new IllegalStateException(label + " " + expected + "!=" + actual); }
  }
  /***********************************************************************/
  public static void assertFileDoesNotExist(String label, String fileName)
  {
    File file = new File(fileName);
    if (file.exists()) { throw new IllegalStateException(label + " file " + file.getAbsolutePath()
        + " already exists"); }
  }
  /***********************************************************************/
  public static void assertFileExists(String label, String fileName)
  {
    File file = new File(fileName);
    if (!file.exists()) { throw new IllegalStateException(label + " file " + file.getAbsolutePath()
        + " does not exist"); }
  }
  /***********************************************************************/
  public static void equals(String label, double expected, double actual, double percission)
  {
    if (!NumberUtils.equals(expected, actual, percission)) { throw new IllegalStateException(label + " "
        + expected + "!=" + actual); }
  }
  /***********************************************************************/
  public static void notEquals(String label, double expected, double actual)
  {
    if (actual == expected) { throw new IllegalStateException(label + " " + expected + "!=" + actual); }
  }
  /***********************************************************************/
  public static void assertEqual(String label, Object expected, Object actual)
  {
    if (!ObjectUtils.isEqual(expected, actual)) { throw new IllegalStateException(label + " " + expected + "!="
        + actual); }
  }
  /***********************************************************************/
  public static void assertNotEqual(String label, Object expected, Object actual)
  {
    if (ObjectUtils.isEqual(expected, actual)) { throw new IllegalStateException(label + " can't = " + expected); }
  }
  /***********************************************************************/
  public static void assertEqual(String label, double expected, double actual, double precision)
  {
    if (!NumberUtils.equals(expected, actual, precision)) { throw new IllegalStateException(label + " " + expected
        + "!=" + actual + "within delta of" + precision); }
  }
  /***********************************************************************/
  public static void assertStartsWith(String label, String haystack, String needle)
  {
    if (haystack == null || !haystack.startsWith(needle)) { throw new IllegalStateException(label + " " + haystack
        + " does not start with " + needle); }
  }
  /***********************************************************************/
  /***********************************************************************/
  public static void assertTrue(String label, boolean b)
  {
    if (!b) { throw new IllegalStateException(label + " was false"); }
  }
}
