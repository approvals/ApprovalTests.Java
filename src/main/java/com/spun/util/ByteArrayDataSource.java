package com.spun.util;

/*
 * @(#)ByteArrayDataSource.java	1.1 00/01/30
 *
 * Copyright 1998-2000 Sun Microsystems, Inc. All Rights Reserved.
 * 
 * This software is the proprietary information of Sun Microsystems, Inc.  
 * Use is subject to license terms.
 * 
 */
import java.io.*;
import javax.activation.*;

/**
 * Comment By LLewellyn: This is a needed class for using mail. For some reason wasn't 
 * made as part of the standard mail package.<P>
 *
 * A simple DataSource for demonstration purposes.
 * This class implements a DataSource from:
 *   an InputStream
 *  a byte array
 *   a String
 *
 * @author John Mani
 * @author Bill Shannon
 * @author Max Spivak
 **/
public class ByteArrayDataSource implements DataSource
{
  private byte[] data;          // data
  private String type;          // content-type
  private String name = "dummy";
  /***********************************************************************/
  /* Create a DataSource from an input stream */
  public ByteArrayDataSource(InputStream is, String type)
  {
    this(is, type, null);
  }
  /***********************************************************************/
  /* Create a DataSource from an input stream */
  public ByteArrayDataSource(InputStream is, String type, String name)
  {
    this.type = type;
    setName(name);
    try
    {
      ByteArrayOutputStream os = new ByteArrayOutputStream();
      int ch;
      while ((ch = is.read()) != -1)
      {
        // XXX - must be made more efficient by
        // doing buffered reads, rather than one byte reads
        os.write(ch);
      }
      data = os.toByteArray();
    }
    catch (IOException ioex)
    {
    }
  }
  /***********************************************************************/
  /* Create a DataSource from a byte array */
  public ByteArrayDataSource(byte[] data, String type)
  {
    this.data = data;
    this.type = type;
  }
  /***********************************************************************/
  public void setName(String name)
  {
    this.name = (name == null) ? "dummy" : name;
  }
  /***********************************************************************/
  /* Create a DataSource from a String */
  public ByteArrayDataSource(String data, String type)
  {
    this(data, type, null);
  }
  /***********************************************************************/
  /* Create a DataSource from a String */
  public ByteArrayDataSource(String data, String type, String name)
  {
    setName(name);
    try
    {
      // Assumption that the string contains only ASCII
      // characters!  Otherwise just pass a charset into this
      // constructor and use it in getBytes()
      this.data = data.getBytes("iso-8859-1");
    }
    catch (UnsupportedEncodingException uex)
    {
    }
    this.type = type;
  }
  /***********************************************************************/
  /**
   * Return an InputStream for the data.
   * Note - a new stream must be returned each time.
   */
  public InputStream getInputStream() throws IOException
  {
    if (data == null) { throw new IOException("no data"); }
    return new ByteArrayInputStream(data);
  }
  /***********************************************************************/
  public OutputStream getOutputStream() throws IOException
  {
    throw new IOException("cannot do this");
  }
  /***********************************************************************/
  public String getContentType()
  {
    return type;
  }
  /***********************************************************************/
  public String getName()
  {
    return name;
  }
  /***********************************************************************/
  /***********************************************************************/
}