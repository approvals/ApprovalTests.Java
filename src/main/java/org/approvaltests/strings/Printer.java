package org.approvaltests.strings;

import org.lambda.functions.implementations.F1;

public class Printer<T> extends F1<T, String>
{
  public Printer(T a, Object... extraVariables)
  {
    super(a, extraVariables);
  }
  public void multiline(String from, String to)
  {
    ret(from + "\r\n => \r\n" + to + "\r\n" + "------------------------------------");
  }
  public static String printLine(Object from, Object to)
  {
    return from + " => " + to + "\r\n";
  }
  public void print(Object from, Object to)
  {
    ret(from + " => " + to);
  }
  public void format(String format, Object... params)
  {
    ret(String.format(format, params));
  }
}
