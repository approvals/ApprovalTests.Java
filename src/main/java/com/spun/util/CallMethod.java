package com.spun.util;

import java.util.ArrayList;

public class CallMethod
{
  private final Class<? extends Object>             clazz;
  private ArrayList<String>                         methodNames = new ArrayList<String>();
  private ArrayList<MethodExecutionPath.Parameters> params      = new ArrayList<MethodExecutionPath.Parameters>();
  public CallMethod(Class<? extends Object> clazz)
  {
    this.clazz = clazz;
  }
  public CallMethod method(String methodName, Object... parameters)
  {
    methodNames.add(methodName);
    if (!ArrayUtils.isEmpty(parameters))
    {
      params.add(new MethodExecutionPath.Parameters(parameters));
    }
    else {
      params.add(new MethodExecutionPath.Parameters((Object[])null));
    }
    return this;
  }
  public MethodExecutionPath go()
  {
    return new MethodExecutionPath(clazz, methodNames.toArray(new String[0]), params.toArray(new MethodExecutionPath.Parameters[0]));
  }
}
