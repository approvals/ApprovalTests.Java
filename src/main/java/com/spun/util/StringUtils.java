package com.spun.util;

import java.io.ByteArrayInputStream;
import java.io.InputStream;
import java.io.UnsupportedEncodingException;
import java.lang.reflect.Field;
import java.util.Arrays;
import java.util.List;
import java.util.Map;
import java.util.Properties;
import java.util.Vector;

import org.lambda.functions.implementations.F1;
import org.lambda.query.Query;

/**
 * A static class of convenience functions for database access
 **/
public class StringUtils
{
  public static final String NEW_LINE = System.getProperty("line.separator");
  /***********************************************************************/
  public static String toURLEncode(String input)
  {
    try
    {
      return input == null ? null : java.net.URLEncoder.encode(input, "UTF-8");
    }
    catch (UnsupportedEncodingException e)
    {
      throw ObjectUtils.throwAsError(e);
    }
  }
  /***********************************************************************/
  public static String loadNullableString(String i)
  {
    return StringUtils.isNonZero(i) ? i.trim() : null;
  }
  /***********************************************************************/
  /**
   * Decode a URLEncoded <code>input</code> String.
   * <p>
   * 
   * If <code>input</code> is <code>null</code>, <code>URLEncode()</code> will
   * return <code>null</code>.
   * 
   * @see java.net.URLDecoder
   */
  public static String URLDecode(String input)
  {
    try
    {
      return input == null ? null : java.net.URLDecoder.decode(input, "UTF-8");
    }
    catch (Exception e)
    {
      return null;
    }
  }
  /************************************************************************/
  public static String[] split(String string, String splitOn)
  {
    return split(string, splitOn, true);
  }
  /************************************************************************/
  public static String[] split(String string, String splitOn, boolean trim)
  {
    if ((string == null) || (splitOn == null) || (splitOn.length() < 1)) { return null; }
    Vector<String> temp = new Vector<String>();
    int length = splitOn.length();
    int start = 0;
    int next = 0;
    while (next != -1)
    {
      String word = null;
      next = string.indexOf(splitOn, start);
      if (next == -1)
      {
        word = string.substring(start);
      }
      else
      {
        word = string.substring(start, next);
        start = next + length;
      }
      temp.add(trim ? word.trim() : word);
    }
    return toArray(temp);
  }
  /************************************************************************/
  public static String replace(String string, String find, String replace)
  {
    if ((string == null) || (find == null) || (replace == null)) { throw new NullPointerException(String.format(
        "[string,find,replace] = [%s,%s,%s]", string, find, replace)); }
    String parts[] = split(string, find, false);
    if (parts.length == 1) { return string; }
    String result = "";
    for (int i = 0; i < parts.length; i++)
    {
      result += parts[i] + replace;
    }
    return result.substring(0, result.length() - replace.length());
  }
  /***********************************************************************/
  public static String stripWhiteSpace(String text)
  {
    return stripWhiteSpace(text, false);
  }
  /***********************************************************************/
  public static String padNumber(long number, int digits)
  {
    String text = "" + number;
    while (text.length() < digits)
    {
      text = "0" + text;
    }
    return text;
  }
  /***********************************************************************/
  public static String padNumber(int number, int digits)
  {
    return padNumber((long) number, digits);
  }
  /***********************************************************************/
  public static String stripWhiteSpace(String text, boolean all)
  {
    StringBuffer newText = new StringBuffer();
    boolean whitespace = false;
    int num = text.length();
    char whiteSpaceChar = ' ';
    boolean atStart = true;
    for (int i = 0; i < num; i++)
    {
      char c = text.charAt(i);
      switch (c)
      {
        case '\r' :
        case '\n' :
          whiteSpaceChar = '\n';
          whitespace = true;
          break;
        case '\t' :
        case ' ' :
          whitespace = true;
          break;
        default :
          if (whitespace && atStart)
          {
            whitespace = false;
          }
          if (whitespace)
          {
            whitespace = false;
            newText.append(all ? '_' : whiteSpaceChar);
            whiteSpaceChar = ' ';
          }
          atStart = false;
          newText.append(c);
          break;
      }
    }
    return newText.toString();
  }
  /************************************************************************/
  /**
   * Turns "mr. frank m Peter" into "Mr. Frank M Peter"
   **/
  public static String toNameUpperCase(String name)
  {
    if (name == null) { return null; }
    StringBuffer returning = new StringBuffer(name.length());
    String upper = name.toUpperCase();
    int place = 0;
    while (place < name.length())
    {
      char letter = name.charAt(place);
      if (letter == ' ')
      {
        returning.append(' ');
        if ((place + 1) < name.length())
        {
          returning.append(upper.charAt(++place));
        }
      }
      else if ((place == 0))
      {
        returning.append(upper.charAt(0));
      }
      else
      {
        returning.append(letter);
      }
      place++;
    }
    return returning.toString();
  }
  /************************************************************************/
  public static String toConvertCamelCaseString(String varName, String insertBeforeCaps)
  {
    StringBuffer staticVarName = new StringBuffer();
    for (int i = 0; i < varName.length(); i++)
    {
      char letter = varName.charAt(i);
      if ((i > 0) && (letter == Character.toUpperCase(letter)) && Character.isLetter(letter))
      {
        staticVarName.append(insertBeforeCaps);
      }
      staticVarName.append(letter);
    }
    return staticVarName.toString();
  }
  /************************************************************************/
  /**
   * Turns "ATTRIUBE_NAME" into "AttributeName"
   **/
  public static String toMethodNameCase(String name)
  {
    StringBuffer returning = new StringBuffer(name.length());
    String upper = name.toUpperCase();
    String lower = name.toLowerCase();
    int place = 0;
    while (place < name.length())
    {
      char letter = lower.charAt(place);
      if (letter == ' ' || letter == '_')
      {
        if ((place + 1) < name.length())
        {
          returning.append(upper.charAt(++place));
        }
      }
      else if ((place == 0))
      {
        returning.append(upper.charAt(0));
      }
      else
      {
        returning.append(letter);
      }
      place++;
    }
    return returning.toString();
  }
  /************************************************************************/
  public static String stripNonNumeric(String number)
  {
    return stripNonNumeric(number, false, false);
  }
  /************************************************************************/
  public static String stripCharacters(String dirtyString, String toStrip)
  {
    StringBuffer cleanStringBuffer = new StringBuffer();
    for (int i = 0; i < dirtyString.length(); i++)
    {
      char currentChar = dirtyString.charAt(i);
      if (toStrip.indexOf(currentChar) == -1)
      {
        cleanStringBuffer.append(currentChar);
      }
    }
    return cleanStringBuffer.toString();
  }
  public static String escapeForXml(String string)
  {
    return string.replaceAll("&", "&amp;");
  }
  /************************************************************************/
  public static String stripNonNumeric(String number, boolean allowDecimal, boolean allowNegative)
  {
    boolean allowExponential = allowDecimal;
    boolean afterE = false;
    if (number == null) { return ""; }
    StringBuffer result = new StringBuffer();
    for (int i = 0; i < number.length(); i++)
    {
      char c = number.charAt(i);
      switch (c)
      {
        case '0' :
        case '1' :
        case '2' :
        case '3' :
        case '4' :
        case '5' :
        case '6' :
        case '7' :
        case '8' :
        case '9' :
          result.append(c);
          afterE = false;
          break;
        case '-' :
          if (allowNegative || afterE)
          {
            result.append(c);
            allowNegative = false;
          }
          break;
        case '.' :
          if (allowDecimal)
          {
            result.append(c);
            allowDecimal = false;
          }
          afterE = false;
          break;
        case 'e' :
        case 'E' :
          if (allowExponential)
          {
            result.append(c);
            allowExponential = false;
          }
          afterE = true;
          break;
        default :
          break;
      }
    }
    return result.toString();
  }
  /***********************************************************************/
  /**
   * A convenience function to check that a String has at least 1 character.
   * 
   * @param string
   *            The string in question
   * @return true if Non Zero.
   **/
  public static boolean isNonZero(String string)
  {
    return ((string != null) && string.trim().length() > 0);
  }
  /***********************************************************************/
  public static boolean isEmpty(String string)
  {
    return !isNonZero(string);
  }
  /************************************************************************/
  /**
   * A convenience function to turn a vector of String objects into an Array
   * of the String objects.
   * 
   * @param vectorOf
   *            a Vector of String objects
   * @return the array of String.
   * @throws Error
   *             if an element of vectorOf is not a String object.
   **/
  public static String[] toArray(java.util.Collection<String> vectorOf)
  {
    if (vectorOf == null) { return new String[0]; }
    String array[] = new String[vectorOf.size()];
    java.util.Iterator<String> iterator = vectorOf.iterator();
    int i = 0;
    while (iterator.hasNext())
    {
      java.lang.Object rowObject = iterator.next();
      if (rowObject instanceof String)
      {
        array[i++] = (String) rowObject;
      }
      else
      {
        throw new Error("toArray[" + i + "] is not an instance of String but a "
            + ObjectUtils.getClassName(rowObject));
      }
    }
    return array;
  }
  /***********************************************************************/
  public static int resolveEnumeration(String value, String[] enumeration)
  {
    return resolveEnumeration(value, enumeration, false);
  }
  /***********************************************************************/
  public static int resolveEnumeration(String value, String[] enumeration, boolean force)
  {
    for (int i = 0; i < enumeration.length; i++)
    {
      if (enumeration[i].equals(value)) { return i; }
    }
    if (force) { throw new Error("Enumeration '" + value + "' not in " + Arrays.asList(enumeration).toString()); }
    return -1;
  }
  /***********************************************************************/
  public static String truncate(String string, int maxLength)
  {
    if (string == null) { return null; }
    return (string.length() <= maxLength) ? string : string.substring(0, maxLength);
  }
  /***********************************************************************/
  public static boolean hasNumeric(String teamId)
  {
    return isNonZero(stripNonNumeric(teamId, false, false));
  }
  /***********************************************************************/
  public static String toHTMLEncode(String string)
  {
    if (string == null) { return null; }
    string = string.replaceAll("<", "&lt;");
    string = string.replaceAll("\n", "<br />");
    return string;
  }
  /***********************************************************************/
  public static String toJavaScriptEncode(String string)
  {
    if (string == null) { return "null"; }
    string = string.replaceAll("\"", "\\\\\"");
    string = string.replaceAll("\r", "\\\\r");
    string = string.replaceAll("\n", "\\\\n");
    return "\"" + string + "\"";
  }
  /***********************************************************************/
  public static boolean isIn(String target, String... fromList)
  {
    return Arrays.asList(fromList).contains(target);
  }
  /***********************************************************************/
  public static boolean isIn(String target, String[] fromList, boolean allowNulls)
  {
    return (target == null && allowNulls) ? true : isIn(target, fromList);
  }
  /************************************************************************/
  public static void assertIn(String target, String[] fromList, boolean allowNulls)
  {
    boolean valid = isIn(target, fromList, allowNulls);
    if (!valid) { throw new IllegalArgumentException("The value '" + target + "' not in "
        + Arrays.asList(fromList)); }
  }
  public static void assertIn(String target, boolean allowNulls, String... options)
  {
    assertIn(target, options, allowNulls);
  }
  /***********************************************************************/
  public static String convertEnumeration(final Object forValue, Class<?> clazz) throws Exception
  {
    F1<Field, Boolean> f1 = new F1<Field, Boolean>(null, forValue)
    {
      {
        ret(ClassUtils.IsPublicStatic(a) && a.get(null).equals(forValue));
      }
    };
    List<Field> fields = Query.where(clazz.getFields(), f1);
    if (fields.isEmpty())
    {
      return "unknown Type " + forValue;
    }
    else
    {
      return fields.get(0).getName();
    }
  }
  /************************************************************************/
  /**
   * 'Tom S Hardy' -> 'Tom S' - 'Hardy'
   **/
  public static String[] splitName(String fullName)
  {
    String names[] = {null, null};
    int split = fullName.lastIndexOf(' ');
    if (split != -1)
    {
      names[0] = fullName.substring(0, split);
      names[1] = fullName.substring(split);
    }
    else
    {
      names[1] = fullName;
    }
    return names;
  }
  /***********************************************************************/
  public static Properties createProperties(String[] properties)
  {
    Properties props = new Properties();
    if (properties == null) { return props; }
    if (properties.length % 2 != 0) { throw new Error("number of strings must be even. found ["
        + properties.length + "] = " + Arrays.asList(properties)); }
    for (int i = 0; i < properties.length; i += 2)
    {
      if (properties[i + 1] != null)
      {
        props.setProperty(properties[i], properties[i + 1]);
      }
      else
      {
        props.setProperty(properties[i], "");
      }
    }
    return props;
  }
  /***********************************************************************/
  public static <T> String toString(String name, T[] array)
  {
    StringBuffer buffer = new StringBuffer();
    name = (name == null ? "array" : name);
    if (array == null || array.length == 0)
    {
      buffer.append(name + ".length = 0");
    }
    else
    {
      int maxPadding = ("" + array.length).length();
      for (int i = 0; i < array.length; i++)
      {
        buffer.append(name + "[" + padNumber(i, maxPadding) + "] = " + array[i] + "\r\n");
      }
    }
    return buffer.toString();
  }
  /***********************************************************************/
  public static <T> String toString(String name, Iterable<T> array)
  {
    StringBuffer buffer = new StringBuffer();
    name = (name == null ? "array" : name);
    if (array != null)
    {
      int count = 0;
      for (T t : array)
      {
        buffer.append(name + "[" + count + "] = " + arrayStringHelper(t) + "\r\n");
        count++;
      }
    }
    if (buffer.length() == 0)
    {
      buffer.append(name + ".length = 0");
    }
    return buffer.toString();
  }
  public static String arrayStringHelper(Object o)
  {
    if (o == null) { return "null"; }
    return o.getClass().isArray() ? Arrays.toString((Object[]) o) : o.toString();
  }
  /***********************************************************************/
  public static String getFirstName(String fullName)
  {
    return splitName(fullName)[0];
  }
  /***********************************************************************/
  public static String getLastName(String fullName)
  {
    return splitName(fullName)[1];
  }
  /************************************************************************/
  public static boolean isLengthWithin(String string, int length)
  {
    return string == null || string.length() <= length;
  }
  /************************************************************************/
  public static boolean equalsIgnoreCase(String one, String two)
  {
    return (one == two || (one != null && one.equalsIgnoreCase(two)));
  }
  /***********************************************************************/
  public static StringBuffer trim(StringBuffer buffer, int i)
  {
    if (buffer == null || buffer.length() < i) { return buffer; }
    buffer.setLength(buffer.length() - i);
    return buffer;
  }
  /***********************************************************************/
  public static String join(String first, String joinBy, String second)
  {
    return isEmpty(second) ? first : first + joinBy + second;
  }
  /***********************************************************************/
  /***********************************************************************/
  public static InputStream convertToInputStream(String string)
  {
    return new ByteArrayInputStream(string.getBytes());
  }
  public static String toString(Map map)
  {
    StringBuffer b = new StringBuffer();
    Object keySet[] = map.keySet().toArray();
    Arrays.sort(keySet);
    for (Object key : keySet)
    {
      b.append(String.format("%s : %s \r\n", key, map.get(key)));
    }
    return b.toString();
  }
}