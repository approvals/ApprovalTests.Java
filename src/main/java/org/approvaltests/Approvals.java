package org.approvaltests;

import java.awt.Component;
import java.awt.Image;
import java.awt.image.BufferedImage;
import java.io.File;
import java.io.FileFilter;
import java.io.FilenameFilter;
import java.sql.ResultSet;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.Map;

import org.apache.hadoop.fs.Path;
import org.approvaltests.ReporterFactory.FileTypes;
import org.approvaltests.approvers.FileApprover;
import org.approvaltests.core.ApprovalFailureOverrider;
import org.approvaltests.core.ApprovalFailureReporter;
import org.approvaltests.core.ApprovalWriter;
import org.approvaltests.namer.ApprovalNamer;
import org.approvaltests.namer.ApprovalNamerFactory;
import org.approvaltests.namer.JUnitStackTraceNamer;
import org.approvaltests.reporters.ExecutableQueryFailure;
import org.approvaltests.writers.ApprovalBinaryFileWriter;
import org.approvaltests.writers.ApprovalTextWriter;
import org.approvaltests.writers.ApprovalXmlWriter;
import org.approvaltests.writers.ComponentApprovalWriter;
import org.approvaltests.writers.DirectoryToDirectoryWriter;
import org.approvaltests.writers.FileApprovalWriter;
import org.approvaltests.writers.ImageApprovalWriter;
import org.approvaltests.writers.ResultSetApprovalWriter;
import org.jrack.RackResponse;
import org.jrack.RackResponseUtils;
import org.lambda.functions.Function1;
import org.lambda.functions.implementations.F1;
import org.lambda.query.Query;

import com.spun.util.ArrayUtils;
import com.spun.util.ObjectUtils;
import com.spun.util.StringUtils;
import com.spun.util.images.ImageWriter;
import com.spun.util.persistence.ExecutableQuery;
import com.spun.util.persistence.SqlLoader;

public class Approvals
{
  public static void verify(String response) throws Exception
  {
    verify(new ApprovalTextWriter(response, "txt"), FileTypes.Text);
  }
  public static void verify(Object o) throws Exception
  {
    verify("" + o);
  }
  public static <T> void verifyAll(String label, T[] array) throws Exception
  {
    verify(new ApprovalTextWriter(StringUtils.toString(label, array), "txt"), FileTypes.Text);
  }
  public static <T> void verifyAll(String header, String label, T[] array) throws Exception
  {
    verify(new ApprovalTextWriter(formatHeader(header) + StringUtils.toString(label, array), "txt"),
        FileTypes.Text);
  }
  public static <T> void verifyAll(T[] values, Function1<T, String> f1)
  {
    String text = ArrayUtils.toString(values, f1);
    verify(new ApprovalTextWriter(text, "txt"), FileTypes.Text);
  }
  public static <T> void verifyAll(String header, T[] values, Function1<T, String> f1)
  {
    verifyAll(header, Arrays.asList(values), f1);
  }
  public static <T> void verifyAll(String header, Iterable<T> array, Function1<T, String> f1)
  {
    String text = formatHeader(header) + ArrayUtils.toString(array, f1);
    verify(new ApprovalTextWriter(text, "txt"), FileTypes.Text);
  }
  private static String formatHeader(String header)
  {
    return StringUtils.isEmpty(header) ? "" : header + "\r\n\r\n\r\n";
  }
  public static <T> void verifyAll(String label, Iterable<T> array) throws Exception
  {
    verify(new ApprovalTextWriter(StringUtils.toString(label, array), "txt"), FileTypes.Text);
  }
  public static <T> void verifyAll(String header, String label, Iterable<T> array) throws Exception
  {
    verify(new ApprovalTextWriter(formatHeader(header) + StringUtils.toString(label, array), "txt"),
        FileTypes.Text);
  }
  public static void verify(Component c) throws Exception
  {
    verify(new ComponentApprovalWriter(c), FileTypes.Image);
  }
  public static void verifyHtml(String response) throws Exception
  {
    verify(new ApprovalTextWriter(response, "html"), FileTypes.Html);
  }
  public static void verify(File generateFile)
  {
    verify(new FileApprovalWriter(generateFile), FileTypes.File);
  }
  public static void verify(Image image)
  {
    approve(ImageWriter.toBufferedImage(image), createApprovalNamer());
  }
  public static void verify(BufferedImage bufferedImage)
  {
    verify(new ImageApprovalWriter(bufferedImage), FileTypes.Image);
  }
  public static void verify(ApprovalWriter writter, ApprovalNamer namer, ApprovalFailureReporter reporter)
  {
    verify(new FileApprover(writter, namer), reporter);
  }
  public static void verify(ApprovalWriter writter, String fileType)
  {
    verify(writter, createApprovalNamer(), ReporterFactory.get(fileType));
  }
  public static void verifyXml(String xml) throws Exception
  {
    verify(new ApprovalXmlWriter(xml), FileTypes.Text);
  }
  public static void verify(FileApprover approver, ApprovalFailureReporter reporter)
  {
    try
    {
      if (!approver.approve())
      {
        boolean passed = false;
        if (reporter instanceof ApprovalFailureOverrider)
        {
          passed = approver.askToChangeReceivedToApproved((ApprovalFailureOverrider) reporter);
        }
        if (!passed)
        {
          approver.reportFailure(reporter);
          approver.fail();
        }
        else
        {
          approver.cleanUpAfterSuccess(reporter);
        }
      }
      else
      {
        approver.cleanUpAfterSuccess(reporter);
      }
    }
    catch (Exception e)
    {
      throw ObjectUtils.throwAsError(e);
    }
  }
  public static void verify(ExecutableQuery query) throws Exception
  {
    verify(new ApprovalTextWriter(query.getQuery(), "txt"), createApprovalNamer(), new ExecutableQueryFailure(
        query));
  }
  public static void verify(Map map) throws Exception
  {
    verify(new ApprovalTextWriter(StringUtils.toString(map), "txt"), FileTypes.Text);
  }
  public static void verify(RackResponse response) throws Exception
  {
    if (isImage(response))
    {
      String fileType = "png";
      verify(new ApprovalBinaryFileWriter(response.getResponse(), fileType), fileType);
    }
    else
    {
      verifyHtml(response.getResponse().toString());
    }
  }
  public static void verify(ResultSet rs) throws Exception
  {
    verify(new ResultSetApprovalWriter(rs), "csv");
  }
  public static void verify(SqlLoader loader) throws Exception
  {
    verify(new SqlLoader.ExecutableWrapper(loader));
  }
  private static boolean isImage(RackResponse response)
  {
    String type = response.getHeaders().get(RackResponseUtils.CONTENT_TYPE);
    return RackResponseUtils.CONTENT_TYPE_IMAGE.equals(type);
  }
  public static ApprovalNamer createApprovalNamer()
  {
    return ApprovalNamerFactory.getApprovalNamer();
  }
  private static void approve(BufferedImage bufferedImage, ApprovalNamer namer)
  {
    verify(new ImageApprovalWriter(bufferedImage), FileTypes.Image);
  }
  public static void verifyEachFileInDirectory(File directory)
  {
    verifyEachFileAgainstMasterDirectory(directory.listFiles());
  }
  public static void verifyEachFileInDirectory(File directory, FileFilter filter)
  {
    verifyEachFileAgainstMasterDirectory(directory.listFiles(filter));
  }
  public static void verifyEachFileInDirectory(File directory, FilenameFilter filter)
  {
    verifyEachFileAgainstMasterDirectory(directory.listFiles(filter));
  }
  private static void verifyEachFileAgainstMasterDirectory(File[] files) throws Error
  {
    ApprovalNamer namer = createApprovalNamer();
    String dirName = namer.getApprovalFileBasePath() + Path.SEPARATOR + namer.getApprovalName() + ".Files";
    File approvedDirectory = new File(dirName);
    List<File> mismatched = new ArrayList<File>();
    for (File f : files)
    {
      if (!f.isDirectory())
      {
        try
        {
          verify(new DirectoryToDirectoryWriter(f, approvedDirectory), FileTypes.File);
        }
        catch (Throwable e)
        {
          mismatched.add(f);
        }
      }
    }
    if (!mismatched.isEmpty())
    {
      String message = "The Following Files did not match up: " + getFileNameList(mismatched);
      throw new Error(message);
    }
  }
  private static String getFileNameList(List<File> mismatched)
  {
    return Query.select(mismatched, new F1<File, String>(mismatched.get(1))
    {
      {
        ret(a.getName());
      }
    }).toString();
  }
}