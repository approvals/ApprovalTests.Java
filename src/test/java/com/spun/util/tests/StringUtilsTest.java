package com.spun.util.tests;

import java.util.Arrays;

import junit.framework.TestCase;

import org.approvaltests.Approvals;
import org.approvaltests.strings.Printer;

import com.spun.util.StringUtils;

public class StringUtilsTest extends TestCase
{

  ReplaceUseCase[] replaceUseCases = {new ReplaceUseCase("quick brown fox", "brown", "white", "quick white fox"),
      new ReplaceUseCase("quick brown quick fox brown", "quick", "slow", "slow brown slow fox brown"),};
  /***********************************************************************/
  public void testJoin() throws Exception
  {
    assertEquals("Falco,Llewellyn", StringUtils.join("Falco", ",", "Llewellyn"));
    assertEquals("Falco", StringUtils.join("Falco", ",", null));
    assertEquals("Falco", StringUtils.join("Falco", ",", ""));
  }
  /***********************************************************************/
  public void testToNameCase()
  {
    assertEquals("Name changed", "Mr. Frank M Peter", StringUtils.toNameUpperCase("mr. frank m peter"));
  }
  /***********************************************************************/
  public void testSplit()
  {
    SplitUseCase[] split = {new SplitUseCase("quick brown fox", " "),
        new SplitUseCase("quick/brown/ fox", "/"),
        new SplitUseCase("quick**brown**fox", "**"),
        new SplitUseCase(" quick   brown fox ", " "),
        new SplitUseCase("quick brown fox", "brown"),};

    Approvals.verifyAll(split, new Printer<SplitUseCase>(split[0]){{format("'%s'.split(%s) => %s", a.start,a.splitOn, Arrays.toString(StringUtils.split(a.start, a.splitOn)));}});
  }
 
  public void testJavaScript()
  {
    String[] strings = {"\"\r\n", "this is a note \"\'\r\nanother liner"};
    Approvals.verifyAll(strings, new Printer<String>(""){{multiline(a,StringUtils.toJavaScriptEncode(a));}});
  }
  /***********************************************************************/
  public void testReplace()
  {
    for (int i = 0; i < replaceUseCases.length; i++)
    {
      assertEquals("Replace failed", StringUtils.replace(replaceUseCases[i].startingString,
          replaceUseCases[i].find, replaceUseCases[i].replace), replaceUseCases[i].expectedString);
    }
  }
  public void testWriteToString() throws Exception
  {
    Approvals.verify(StringUtils.toString("things", new Object[]{1, null, "hi"}));
  }
  /***********************************************************************/
  public class SplitUseCase
  {
    String   start;
    String   splitOn;
    String[] expectedArray;
    public SplitUseCase(String startingString, String splitOn, String... expectedArray)
    {
      this.start = startingString;
      this.splitOn = splitOn;
      this.expectedArray = expectedArray;
    }
  }
  /************************************************************************/
  public class ReplaceUseCase
  {
    String startingString;
    String find;
    String replace;
    String expectedString;
    public ReplaceUseCase(String startingString, String find, String replace, String expectedString)
    {
      this.startingString = startingString;
      this.find = find;
      this.replace = replace;
      this.expectedString = expectedString;
    }
  }
  /***********************************************************************/
  /***********************************************************************/
}
