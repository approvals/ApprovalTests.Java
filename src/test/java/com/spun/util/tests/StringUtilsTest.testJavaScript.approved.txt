"

 => 
"\"\r\n"
------------------------------------
this is a note "'
another liner
 => 
"this is a note \"'\r\nanother liner"
------------------------------------
